core/templates/dev/head/domain/skill/ConceptCardBackendApiService.ts

```diff
@@ -86,35 +91,48 @@ oppia.factory('ConceptCardBackendApiService', [
        */
       loadConceptCards: function(skillIds) {
         return $q(function(resolve, reject) {
-          if (_isCached(skillIds)) {
+          var uncachedSkillIds = _getUncachedSkillIds(skillIds);
+          var conceptCards = [];
+          // Concept card partial cache hit case.
+          if (uncachedSkillIds.length !== 0) {
+            _fetchConceptCards(
+              uncachedSkillIds, function(uncachedConceptCards) {
+                skillIds.forEach(function(skillId) {
+                  if (uncachedSkillIds.includes(skillId)) {
+                    conceptCards.push(
+                      uncachedConceptCards[uncachedSkillIds.indexOf(skillId)]);
```

Comment:
You can do the caching here itself to avoid looping over once again

Reply:
Do you mean just do `conceptCards = uncachedConceptCards`? If so, I'm doing it in this way to keep skillIds and conceptCards the same order for future use.

Reply:
Oh sorry I just realized what you meant. Did you mean cache the fetched conceptCards in this loop as well?

Reply:
Yeah, I think that's what she meant, so that the next loop can be avoided.

Reply:
Done.